test/bin/opam-provided.t/run.t

```diff
@@ -0,0 +1,125 @@
+We have a package that should not be installed via opam. There might be various
+reasons we might not want that (e.g. it cannot be built with `dune`), it is
+meant as a helper binary etc.
+
+We start with a minimal opam-repository:
+
+  $ gen-minimal-repo
+
+We have a file that depends on package `b`.
+
+  $ opam show --no-lint --raw -fdepends ./vendored.opam
+  "dune" "b"
+
+This package does not set any `opam-provided` packages:
+
+  $ opam show --no-lint --raw -fx-opam-monorepo-opam-provided ./vendored.opam
+  
+
+It should lock successfully.
+
+  $ opam-monorepo lock vendored > /dev/null
+
+The lockfile should thus contain the package `b` and mark it as `vendor` since
```

Comment:
`run.t` files are assumed to be Markdown (I guess?) and should use `` ` ``.

Reply:
That's good to know, as most of them were in [ ], so I followed that formatting. So you're saying things in monospace in `run.t` files should be wrapped in ``` instead of [ ]?

Reply:
Yes, exactly. `[]` is only for "odoc"-markup documents like `.ml` files (source code documentation) and `.mld` files (free text documentation).

Reply:
Thank you for clarifying that. I'll go through `run.t` and fix those.

Reply:
Putting b in "b" instead of single quotes ('b') or backslashes (`b`) to match the code version. Does that work for you?